### DISCUSSION TITLE
How does craco actually work?

### DISCUSSION THREAD

**Discussion by Makistos** [dilanx/craco#472] [General]
I tried looking at the docs but can't figure this out. Does it end up changing the contents of js/tsconfig.json? I'm struggling to get aliases to work and can't figure out what's wrong. --verbose shows no issues (and says "Added webpack alias.") but tsconfig.json doesn't have the changes. More specifically I'm trying to add some path aliases and even though I think I do it like several pages say compilation fails when I try to use them.

So how would I debug this issue?

**Comment by dilanx** [comment]
CRACO doesn't modify any config files and instead modifies the modules at runtime before they're used by your project. The webpack configuration that you wrote in your CRACO config might be incorrect, but feel free to share it here if you'd like.

**Comment by Makistos** [comment]
Seems like I deleted that config file some time back since I couldn't get it to work. I'll see if I have another go later.
